operator/validator/controller.go

```diff
@@ -517,12 +520,16 @@ func (c *controller) StartValidators() {
 	}
 
 	var ownShares []*ssvtypes.SSVShare
-	var allPubKeys = make([][]byte, 0, len(shares))
+	var pubKeysToFetch [][]byte
+	activeSubnets := c.network.ActiveSubnets()
 	for _, share := range shares {
 		if c.operatorDataStore.GetOperatorID() != 0 && share.BelongsToOperator(c.operatorDataStore.GetOperatorID()) {
 			ownShares = append(ownShares, share)
 		}
-		allPubKeys = append(allPubKeys, share.ValidatorPubKey[:])
+		subnet := commons.CommitteeSubnet(share.CommitteeID())
+		if uint64(len(activeSubnets)) >= subnet && activeSubnets[subnet] != 0 {
+			pubKeysToFetch = append(pubKeysToFetch, share.ValidatorPubKey[:])
```

Comment:
Honestly this sounds a bit cycle-ish for me. We are subscribing to validators'/committee's subnet after a metadata fetch that indicates the validator is active_ongoing. then here we are only fetching validators that show up on p2p active subnets?

Have you tested this on a fresh db sync? Am I missing something? @nkryuchkov @moshe-blox 
I feel like we should probably get the list of validators/committees to fetch the database based on the contract sync. And the p2p active subnets should be a result of this fetch.

So 
contract sync -> validator list -> metadata fetch(validator list) -> active subnets (fetched as active_ongoing).

Reply:
@y0sher good point, agree 👌

feels like an "import cycle" 🤣 

@nkryuchkov we should fetch metadata for validators in all subnets of our committees, whether they're active or not (because we don't really know until we fetch)

perhaps instead of calling `c.network.ActiveSubnets()` we should call a func that returns the subnets of all of our committees (whether active or not), and the rest of the logic can remain the same.

i think we might already have this logic in the DynamicMaxPeers PR, so maybe we just need to export it to a func

Reply:
@y0sher @moshe-blox please review if https://github.com/ssvlabs/ssv/pull/1787/commits/fd45bebf3683c7c2c9791f31cefc8536d396c13d is what you meant